src/assets/HTMLAsset.js

```diff
@@ -41,6 +41,28 @@ class HTMLAsset extends Asset {
     this.ast.walk(node => {
       if (node.attrs) {
         for (let attr in node.attrs) {
+          if (node.tag === 'img' && attr === 'srcset') {
+            const newSources = [];
+            for (const source of node.attrs[attr].split(',')) {
+              const pair = source.trim().split(' ');
+              if (pair.length === 0) continue;
+
+              let assetPath = this.addURLDependency(
+                decodeURIComponent(pair[0])
+              );
+              if (!isURL(assetPath)) {
+                assetPath = urlJoin(this.options.publicURL, assetPath);
+              }
+              pair[0] = assetPath;
+
+              newSources.push(pair.join(' '));
+            }
+
+            node.attrs[attr] = newSources.join(',');
+            this.isAstDirty = true;
+
+            continue;
```

Comment:
Could we move this logic for parsing srcset to a separate function? Maybe we could have a map somewhere of custom handlers for certain attrs?

Reply:
I moved this logic to its own function. About custom handlers, perhaps its better to wait until there is at least another one?